### PR TITLE
chore(project): add anchor tags for better seo

### PR DESCRIPTION
**Pull request by olga-jwp**
## Description

[OWA-6](https://jwplayer.atlassian.net/browse/OWA-6)

This PR solves # .

### Steps completed:

- Add <a> tag inside Shelf.tsx

- Remove onCardClick() from Shelflist.tsx

According to our definition of done, I have completed the following steps:

- [ ] Acceptance criteria met
- [ ] Unit tests added
- [ ] Docs updated (including config and env variables)
- [ ] Translations added
- [ ] UX tested
- [ ] Browsers / platforms tested
- [ ] Rebased & ready to merge without conflicts
- [ ] Reviewed own code

[OWA-6]: https://jwplayer.atlassian.net/browse/OWA-6?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Shelf/Shelf.tsx [hunk 1]
```diff
@@ -71,22 +71,24 @@ const Shelf = ({
...
-          onClick={isInView ? () => onCardClick(item, playlist.feedid, type) : undefined}
-          onHover={typeof onCardHover === 'function' ? () => onCardHover(item) : undefined}
-          featured={featured}
-          disabled={!isInView}
-          loading={loading}
-          isLocked={isLocked(accessModel, isLoggedIn, hasSubscription, item)}
-          posterAspect={posterAspect}
-          item={item}
-        />
+        <a href={isInView ? url : undefined} className={styles.cardWrapper}>
```
Comment (ChristiaanScheermeijer) [inline src/components/Shelf/Shelf.tsx]: I think it is better to update the `Card` component because now it results in a button wrapped by an anchor tag. This negatively impacts the accessibility for keyboard/screenreader users. Besides that, the Card component is also used on the playlist screens, which use the `CardGrid` component.

**Current HTML**
```html
<a href="/m/uB8aRnu6/agent-327?r=dGSUzs9o" class="_cardWrapper_1ge8j_66">
  <div class="_card_1lln7_1" tabindex="0" role="button" aria-label="Play Agent 327">
```

**Expected HTML**
```html
<a href="/m/uB8aRnu6/agent-327?r=dGSUzs9o" class="_card_1lln7_1" aria-label="Agent 327">
```

I also see that the `aria-label` contains the prefix "Play " which can be removed.

Comment (olga-jwp) [reply src/components/Shelf/Shelf.tsx]: Hey Chris, thanks! I've discussed that with Danny, and I am indeed moving the tag inside of the Card component.

File: src/components/CardGrid/CardGrid.test.tsx [hunk 2]
```diff
@@ -9,15 +10,17 @@ describe('<CardGrid>', () => {
...
-      <CardGrid
-        playlist={placeholderData}
-        onCardHover={vi.fn()}
-        onCardClick={vi.fn()}
-        isLoading={false}
-        accessModel={'SVOD'}
-        isLoggedIn={true}
-        hasSubscription={true}
-      />,
+      <BrowserRouter>
```
Comment (budzworthy) [inline src/components/CardGrid/CardGrid.test.tsx]: Why did you add this here? I don't think we need another BrowserRouter (in fact multiples might not play nicely together) and we already have 1 at the top level in App.tsx

Comment (budzworthy) [reply src/components/CardGrid/CardGrid.test.tsx]: Oh this is a test file, my bad. You can use the wrapper functions in testUtils.tsx

File: src/components/Card/Card.tsx [hunk 3]
```diff
@@ -28,6 +29,7 @@ type CardProps = {
   isCurrent?: boolean;
   isLocked?: boolean;
   currentLabel?: string;
+  redirectUrl?: string;
```
Comment (budzworthy) [inline src/components/Card/Card.tsx]: Let's just call this `url`. The `redirect` prefix might imply some sort of redirect vs navigation and be a bit confusing

File: src/components/Card/Card.tsx [hunk 4]
```diff
@@ -86,14 +89,15 @@ function Card({
   };
 
   return (
-    <div
+    <Link
+      to={redirectUrl ?? ''}
```
Comment (budzworthy) [inline src/components/Card/Card.tsx]: Is there any reason the url can't be a required prop?

File: src/components/CardGrid/CardGrid.test.tsx [hunk 5]
```diff
@@ -9,15 +10,17 @@ describe('<CardGrid>', () => {
   it('renders and matches snapshot', () => {
     const placeholderData = generatePlaylistPlaceholder();
```
Comment (budzworthy) [inline src/components/CardGrid/CardGrid.test.tsx]: Can you try using playlist fixture data instead of this placeholder? I think then we'll have more data filled into the snapshots like title and ID's: https://github.com/jwplayer/ott-web-app/blob/579cd0cde5d8c48933867ef3371e542252bdbbb7/src/utils/image.test.ts#L9

File: src/components/Card/Card.tsx (hunk 4 above)
Comment (olga-jwp) [reply src/components/Card/Card.tsx]: I set it optional because of this `isInView` flag `url={isInView ? url : undefined}` in Shelf.tsx

File: src/components/Shelf/Shelf.tsx [hunk 6]
```diff
@@ -71,22 +71,23 @@ const Shelf = ({
...
           progress={watchHistory ? watchHistory[item.mediaid] : undefined}
-          onClick={isInView ? () => onCardClick(item, playlist.feedid, type) : undefined}
           onHover={typeof onCardHover === 'function' ? () => onCardHover(item) : undefined}
           featured={featured}
           disabled={!isInView}
           loading={loading}
           isLocked={isLocked(accessModel, isLoggedIn, hasSubscription, item)}
           posterAspect={posterAspect}
           item={item}
+          redirectUrl={isInView ? url : undefined}
```
Comment (budzworthy) [inline src/components/Shelf/Shelf.tsx]: I don't think we still need this isInView condition. iirc it was to prevent accidental clicks of the cards on the edges under the arrows. Hopefully the links work better with the slider? Can you test and verify? We'll get fuller indexing without special logic if we can add urls to all of the cards. Otherwise, we can also add a bot detection check here and set the url if a bot is detected or is in view.

Comment (olga-jwp) [reply src/components/Shelf/Shelf.tsx]: Unfortunately, the behavior for the slider is the same
https://github.com/jwplayer/ott-web-app/assets/118803926/d8cec1c7-df6e-44fe-a707-3a2270af906c

Comment (budzworthy) [reply src/components/Shelf/Shelf.tsx]: OK, then I will create a separate ticket for this w/ bot detection. Thank you for checking!

File: src/components/Card/Card.tsx [hunk 7]
```diff
@@ -28,6 +29,7 @@ type CardProps = {
   isCurrent?: boolean;
   isLocked?: boolean;
   currentLabel?: string;
+  url?: string;
```
Comment (budzworthy) [inline src/components/Card/Card.tsx]: What if we make the url property required, and add a 'disabled' property that we pass to the Link when we don't want it to be clickable? An empty string `to` value causes weird behavior (it drops the entire path besides what was matched by the current React Router route and it creates entries in the nav history.)

I also don't like having optional inputs when we expect a value, because it makes it easy for developers to forget to set properties where the component is used.

File: src/components/Card/Card.tsx (hunk 4 above)
Comment (budzworthy) [reply src/components/Card/Card.tsx]: I did some investigating into what it means to set a link to a blank string, and I don't think it is going to do what you expect it to: https://stackoverflow.com/a/76738446/1345424

I'd suggest you set make url required, always set it, and just add a disabled prop when isInView is false.

File: src/components/Card/Card.test.tsx [hunk 8]
```diff
@@ -11,23 +11,23 @@ const itemWithImage = { title: 'This is a movie', duration: 120, cardImage: 'htt
 
 describe('<Card>', () => {
   it('renders card with video title', () => {
-    const { getByText } = renderWithRouter(<Card item={item} onClick={() => ''} />);
+    const { getByText } = renderWithRouter(<Card item={item} onClick={() => ''} url="" />);
```
Comment (budzworthy) [inline src/components/Card/Card.test.tsx]: Let's put some dummy url in so we see it on the snapshots
```suggestion
    const { getByText } = renderWithRouter(<Card item={item} onClick={() => ''} url="https://test.dummy.jwplayer.com" />);
```

File: src/components/Card/Card.tsx [hunk 9]
```diff
@@ -90,7 +90,7 @@ function Card({
 
   return (
     <Link
-      to={url ?? ''}
+      to={disabled ? '/' : url}
```
Comment (budzworthy) [inline src/components/Card/Card.tsx]: I think this will only work on the home page where we want to route to the root (`/`).  You should be able to set the `to` prop to the url unconditionally, but then set disabled on the Link itself.
```suggestion
      to={url}
      disabled={disabled}
```

File: src/pages/ScreenRouting/playlistScreens/PlaylistGrid/PlaylistGrid.tsx [hunk 10]
```diff
@@ -48,14 +43,7 @@ const PlaylistGrid: ScreenComponent<Playlist> = ({ data, isLoading }) => {
...
       <main className={styles.main}>
-        <CardGrid
-          playlist={filteredPlaylist}
-          onCardClick={onCardClick}
-          accessModel={accessModel}
-          isLoggedIn={!!user}
-          hasSubscription={!!subscription}
-          isLoading={isLoading}
-        />
+        <CardGrid playlist={filteredPlaylist} accessModel={accessModel} isLoggedIn={!!user} hasSubscription={!!subscription} isLoading={isLoading} />
```
Comment (budzworthy) [inline src/pages/ScreenRouting/playlistScreens/PlaylistGrid/PlaylistGrid.tsx]: I usually suggest the layout with 1 property per line unless there are only 1-2 props.

File: src/pages/Search/Search.tsx [hunk 11]
```diff
@@ -44,13 +41,11 @@ const Search = () => {
     }
   }, [firstRender, query, searchQuery, updateSearchQuery]);
 
-  const onCardClick = (playlistItem: PlaylistItem) => {
+  const onCardClick = () => {
     useUIStore.setState({
```
Comment (budzworthy) [inline src/pages/Search/Search.tsx]: Maybe it's cleaner to just do this in on-unmount now that we aren't navigating via card clicks.
https://legacy.reactjs.org/docs/hooks-effect.html#effects-with-cleanup
```suggestion
  useEffect(() => {
    return () => {
      useUIStore.setState({
        searchQuery: '',
        searchActive: false,
      });
    }
  }, []);
```

File: src/pages/ScreenRouting/playlistScreens/PlaylistGrid/PlaylistGrid.tsx (hunk 10 above)
Comment (olga-jwp) [reply src/pages/ScreenRouting/playlistScreens/PlaylistGrid/PlaylistGrid.tsx]: Looks like it's corrected by husky

File: src/components/Card/Card.test.tsx [hunk 12]
```diff
@@ -11,23 +11,23 @@ const itemWithImage = { title: 'This is a movie', duration: 120, cardImage: 'htt
 
 describe('<Card>', () => {
   it('renders card with video title', () => {
-    const { getByText } = renderWithRouter(<Card item={item} onClick={() => ''} />);
+    const { getByText } = renderWithRouter(<Card item={item} onClick={() => ''} url="https://test.dummy.jwplayer.com" />);
```
Comment (budzworthy) [inline src/components/Card/Card.test.tsx]: Whoops, there aren't any snapshot tests to actually check the url part of this. Can you add one for the Card component?

File: src/components/Card/Card.tsx [hunk 13]
```diff
@@ -90,9 +90,9 @@ function Card({
 
   return (
     <Link
-      to={url ?? ''}
+      to={url}
       className={cardClassName}
-      onClick={onClick}
+      onClick={disabled ? (e) => e.preventDefault() : onClick}
```
Comment (budzworthy) [inline src/components/Card/Card.tsx]: Is there anywhere still using the onClick prop or can we get rid of it?

Comment (olga-jwp) [reply src/components/Card/Card.tsx]: https://github.com/jwplayer/ott-web-app/blob/develop/src/components/VideoLayout/VideoLayout.tsx#L126C36-L126C36
That's the only place left

Comment (budzworthy) [reply src/components/Card/Card.tsx]: OK, and that's the one we need to dig into series logic for, right? Can you just add a TODO to remove the onClick prop once we don't need it anymore?

Comment (olga-jwp) [reply src/components/Card/Card.tsx]: Not really, it's for displaying the related videos. I'll check if this can be removed

File: src/components/Card/Card.tsx [hunk 14]
```diff
@@ -90,13 +88,12 @@ function Card({
...
-      to={url ?? ''}
+      to={url}
       className={cardClassName}
-      onClick={onClick}
+      onClick={disabled ? (e) => e.preventDefault() : undefined}
       onMouseEnter={onHover}
       tabIndex={disabled ? -1 : 0}
-      onKeyDown={(event: KeyboardEvent) => (event.key === 'Enter' || event.key === ' ') && !disabled && onClick && onClick()}
-      role="button"
+      onKeyDown={(event: KeyboardEvent) => (event.key === 'Enter' || event.key === ' ') && !disabled}
```
Comment (budzworthy) [inline src/components/Card/Card.tsx]: This method isn't doing anything. I think you can remove it. The disabled links won't be tabbable anyhow.

File: src/components/CardGrid/CardGrid.test.tsx [hunk 15]
```diff
@@ -13,11 +13,11 @@ describe('<CardGrid>', () => {
       <CardGrid
         playlist={playlist}
         onCardHover={vi.fn()}
-        onCardClick={vi.fn()}
         isLoading={false}
         accessModel={'SVOD'}
         isLoggedIn={true}
         hasSubscription={true}
+        getUrl={() => '/https://test.dummy.jwplayer.com'}
```
Comment (budzworthy) [inline src/components/CardGrid/CardGrid.test.tsx]: You can increase the test coverage a little bit by having this function take a playlist item and set the url with a param or something based on it.
```getUrl={(item: PlaylistItem) => `https://test.dummy.jwplayer.com?media_id=${item.id}`}```

File: src/pages/LegacySeries/LegacySeries.tsx [hunk 16]
```diff
@@ -79,8 +79,10 @@ const LegacySeries = () => {
 
   // Handlers
   const goBack = () => episode && navigate(legacySeriesURL({ episodeId: episode.mediaid, seriesId, play: false, playlistId: feedId }));
-  const onCardClick = (toEpisode: PlaylistItem) =>
-    seriesPlaylist && navigate(legacySeriesURL({ episodeId: toEpisode.mediaid, seriesId, play: false, playlistId: feedId }));
+  const getUrl = (toEpisode: PlaylistItem) => {
+    return (seriesPlaylist && legacySeriesURL({ episodeId: toEpisode.mediaid, seriesId, play: false, playlistId: feedId })) || '';
```
Comment (budzworthy) [inline src/pages/LegacySeries/LegacySeries.tsx]:
```suggestion
    return seriesPlaylist ? legacySeriesURL({ episodeId: toEpisode.mediaid, seriesId, play: false, playlistId: feedId })) : '';
```

File: src/pages/ScreenRouting/mediaScreens/MediaSeries/MediaSeries.tsx [hunk 17]
```diff
@@ -182,6 +182,10 @@ const MediaSeries: ScreenComponent<PlaylistItem> = ({ data: seriesMedia }) => {
 
   if (!seriesMedia || !series) return <ErrorPage title={t('series_error')} />;
 
+  const getURL = (toEpisode: PlaylistItem) => {
+    return mediaURL({ media: seriesMedia, episodeId: toEpisode.mediaid });
```
Comment (budzworthy) [inline src/pages/ScreenRouting/mediaScreens/MediaSeries/MediaSeries.tsx]: This one is going to be a little different, but we still want to replace the `onCardClick`method.
```suggestion
    return addQueryParam(location, 'e', toEpisode.mediaid);
```

`location` should come from a useLocation hook, which you can put up by the `useSearchParams` hook.

File: src/pages/ScreenRouting/playlistScreens/PlaylistLiveChannels/PlaylistLiveChannels.tsx [hunk 18]
```diff
@@ -201,6 +201,7 @@ const PlaylistLiveChannels: ScreenComponent<Playlist> = ({ data: { feedid, playl
         shareButton={shareButton}
         trailerButton={null}
         favoriteButton={null}
+        getURL={() => ''} // needs to be added with <Link> to <Epg>
```
Comment (budzworthy) [inline src/pages/ScreenRouting/playlistScreens/PlaylistLiveChannels/PlaylistLiveChannels.tsx]:
```suggestion
        getURL={() => ''} // TODO: OWA-31 - Add link for EPG item
```

File: src/components/CardGrid/CardGrid.test.tsx [hunk 19]
```diff
@@ -13,11 +13,11 @@ describe('<CardGrid>', () => {
       <CardGrid
         playlist={playlist}
         onCardHover={vi.fn()}
-        onCardClick={vi.fn()}
         isLoading={false}
         accessModel={'SVOD'}
         isLoggedIn={true}
         hasSubscription={true}
+        getUrl={(item: PlaylistItem) => `https://test.dummy.jwplayer.com?media_id=${item.id}`}
```
Comment (budzworthy) [inline src/components/CardGrid/CardGrid.test.tsx]: Sorry, this should have been `mediaid`, not just `id`. Did you notice all the undefined values in the snapshot?
```suggestion
        getUrl={(item: PlaylistItem) => `https://test.dummy.jwplayer.com?media_id=${item.mediaid}`}
```

**Review by budzworthy — APPROVED** [review]
